cluster/src/main/java/org/apache/iotdb/cluster/log/RaftLogManager.java

```diff
@@ -0,0 +1,195 @@
+package org.apache.iotdb.cluster.log;
+
+import org.apache.iotdb.cluster.exception.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RaftLogManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(RaftLogManager.class);
+
+    UnCommittedEntryManager unCommittedEntryManager;
+    CommittedEntryManager committedEntryManager;
+    StableEntryManager stableEntryManager;
+
+    long committed;
+    long applied;
+
+    public RaftLogManager(CommittedEntryManager committedEntryManager, StableEntryManager stableEntryManager) {
+        this.committedEntryManager = committedEntryManager;
+        this.stableEntryManager = stableEntryManager;
+        long first = committedEntryManager.getFirstIndex();
+        long last = committedEntryManager.getLastIndex();
+        this.unCommittedEntryManager = new UnCommittedEntryManager(last + 1);
+        this.committed = first - 1;
+        this.applied = first - 1;
+    }
+
+    public long getTerm(long index) {
+        long dummyIndex = getFirstIndex() - 1;
+        if (index < dummyIndex || index > getLastIndex()) {
+            return -1;
+        }
+        long term = unCommittedEntryManager.maybeTerm(index);
+        if (term != -1) {
+            return term;
+        }
+        return committedEntryManager.getTerm(index);
+    }
+
+    public long getFirstIndex() {
+        return committedEntryManager.getFirstIndex();
+    }
+
+    public long getLastIndex() {
+        long last = unCommittedEntryManager.maybeLastIndex();
+        if (last != -1) {
+            return last;
+        }
+        return committedEntryManager.getLastIndex();
+    }
+
+    public long getLastTerm() {
+        long term = getTerm(getLastIndex());
+        if (term == -1) {
+            logger.error("unexpected error when getting the last term");
```

Comment:
Normally this should not happen, because the lastIndex of commitedEntryManager must exist, so its corresponding term also exists.
If it occurs,that can only be an unknown error needing manual inventory.